lib/org/chromium/webidl/Outputer.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'Outputer',
+
+  documentation: 'Outputer for parsed Web IDL',
+
+  properties: [
+    {
+      class: 'String',
+      documentation: 'Buffer for current output run.',
+      name: 'buf_',
+      value: '',
+    },
+    {
+      class: 'Int',
+      documentation: 'Line-based indentation level.',
+      name: 'indentLevel_',
+      value: 0,
+    },
+    {
+      class: 'String',
+      documentation: 'String equivalent of one indentation level.',
+      name: 'indentStr',
+      value: '  ',
+    },
+    {
+      class: 'String',
+      documentation: 'String equivalent of "new line".',
+      name: 'nlStr',
+      value: '\n',
+    },
+    {
+      class: 'Boolean',
+      documentation: 'Toggle for enabling/disabling "pretty printing".',
+      name: 'pretty',
+      value: true,
+      postSet: function(_, p) {
+        if (p) {
+          this.clearProperty('indentStr');
+          this.clearProperty('nlStr');
+        } else {
+          this.indentStr = this.nlStr = null;
+        }
+      },
+    },
+  ],
+
+  methods: [
+    function reset() {
+      this.indentLevel_ = 0;
+      this.buf_ = '';
+      return this;
+    },
+    function out() {
+      for (var i = 0; i < arguments.length; i++) this.buf_ += arguments[i];
```

Comment:
We used to append to an array and then join(), when that was the faster method, but when V8 optimized +=, we switched to the current method as a result of benchmark comparisons.